### DISCUSSION TITLE
Building for MacOS gives error: qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in ""

### DISCUSSION THREAD

**Discussion by coryclemens** [jamulussoftware/jamulus#3425] [Technical Support & Help] [answered]
Trying to work toward contributing to this project because it is a problem I have wanted to solve for a long time.. though, when following instructions on [Compiling](https://github.com/jamulussoftware/jamulus/blob/main/COMPILING.md ), I:
1) Download XCode
2) Install qt@5 via homebrew (it installs it to /opt/homebrew/opt/qt@5) -  I also run the brew link command here
3) After all the building, attempt to execute ./Release/Jamulus.app/Contents/MacOS/Jamulus

And get the following error:
**- allocated port number: 22134**
**qt.qpa.plugin: Could not find the Qt platform plugin "cocoa" in "/opt/homebrew/opt/qt@5/plugins/platforms"**
**This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.**

**zsh: abort      ./Jamulus**

I tried setting some ~/.zshrc paths , specifically 
export QT_QPA_PLATFORM_PLUGIN_PATH=/opt/homebrew/opt/qt@5/plugins/platforms
export QT_PLUGIN_PATH=/opt/homebrew/opt/qt@5/plugins/

But to no luck. I looked this issue up online and none of the solutions have worked.

**Comment by coryclemens** [comment]
Apparently Qt 5.15.x not supported on Apple M1/M2 silicon arm64... if anyone has developed on this platform please get in touch with me, I would love to have some technical support setting up an environment for this.

**Comment by ann0see** [comment, marked as answer]
Hi!
Thanks for reaching out. This is in fact a known issue (https://github.com/jamulussoftware/jamulus/issues/3078). You need to run macdeployqt and probably use Qt6. 
Have a look at how the CI works: https://github.com/jamulussoftware/jamulus/blob/main/.github/autobuild/mac.sh

This reminds me that we need to update the docs for macOS.

**Reply by coryclemens** [reply]
Resolved this by installing Qt6, following the advice here [Discussion 3240](https://github.com/orgs/jamulussoftware/discussions/3240) to build for arm64 only, and then running macdeployqt on Release/Jamulus.app before executing ./Jamulus

**Reply by ann0see** [reply]
That's great!

**Comment by David-Araripe** [comment]
I ran into this issue setting up a similar software. Other than installing `qt@6` as suggested, I had to fix the plugin path like this:
```shell
export QT_QPA_PLATFORM_PLUGIN_PATH=/opt/homebrew/opt/qtbase/share/qt/plugins/platforms/
```
